### PR TITLE
Feature - Speedup docker builds

### PR DESCRIPTION
**Pull request by naveensrinivasan**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Decreased the docker builds duration by using cache

#### Which issue(s) this PR fixes:
None

#### Does this PR have a test?
N/A

#### Does this PR introduce a user-facing change?
NONE

```release-note
 Decreased docker builds duration by using cache
```

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Comment by naveensrinivasan** [comment]
@saschagrunert  How do I know what version of docker is running in prow? Can it be upgraded?

**Comment by naveensrinivasan** [comment]
Decided not to use the new `cache` syntax because prow does not support the latest version of Docker.

**Comment by saschagrunert** [comment]
> @saschagrunert How do I know what version of docker is running in prow? Can it be upgraded?

The prow instance is running inside of the Google Cloud which means we cannot update docker that way.

File: Dockerfile [hunk 1]
```diff
@@ -28,7 +28,12 @@ RUN adduser \
     --uid "${UID}" \
     "${USER}"
 
+# Speed up build by leveraging docker layer caching
+COPY go.mod go.sum ./
+RUN go mod download
```
Comment (saschagrunert) [inline Dockerfile:33]: Do we really need this? We removed it because of the vendoring.

Comment (JAORMX) [reply Dockerfile:33]: yeah, this shouldn't be needed any more.

File: Dockerfile [hunk 2]
```diff
@@ -28,7 +28,9 @@ RUN adduser \
     --uid "${UID}" \
     "${USER}"
 
-ADD . /work
+COPY . /work
```
Comment (saschagrunert) [inline Dockerfile:31]: What's the reason to use `COPY` instead of `ADD` here?

**Review by saschagrunert — COMMENTED** [review]
So if I get it correctly, then we add an additional layer to speedup the build? If that is the case can you please outline that fact in the commit message/body? :upside_down_face:

File: Dockerfile (hunk 2 above)
Comment (naveensrinivasan) [reply Dockerfile:31]: hadolint had these warnings and that is the reason for the change https://github.com/hadolint/hadolint/wiki/DL3020

**Comment by naveensrinivasan** [comment]
> So if I get it correctly, then we add an additional layer to speedup the build? If that is the case can you please outline that fact in the commit message/body? 🙃

I can update the message

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**